packages/components/src/autocomplete/index.tsx

```diff
@@ -163,20 +194,36 @@ export function useAutocomplete( {
 		) {
 			return;
 		}
+
 		switch ( event.key ) {
-			case 'ArrowUp':
-				setSelectedIndex(
+			case 'ArrowUp': {
+				const newIndex =
 					( selectedIndex === 0
 						? filteredOptions.length
-						: selectedIndex ) - 1
-				);
+						: selectedIndex ) - 1;
+				setSelectedIndex( newIndex );
+				if ( isAppleOS() ) {
+					speak(
+						filteredOptions[ newIndex ].textLabel ||
+							getNodeText( filteredOptions[ newIndex ].label ),
+						'assertive'
+					);
+				}
 				break;
+			}
 
-			case 'ArrowDown':
-				setSelectedIndex(
-					( selectedIndex + 1 ) % filteredOptions.length
-				);
+			case 'ArrowDown': {
+				const newIndex = ( selectedIndex + 1 ) % filteredOptions.length;
+				setSelectedIndex( newIndex );
+				if ( isAppleOS() ) {
```

Comment:
I think it would be helpful to have a link to this PR and a brief comment explaining why we need to use `speak` on appleOS. Let's do it in a follow-up though, so we don't need to retrigger all the tests since they're currently passing.